### PR TITLE
Feature: shortcuts

### PR DESCRIPTION
**Pull request by phil1995**
Adds the "Get Folder" and "Save File" intents. This allows the user to create various shortcuts / automations in the Shortcuts app, including an automatic photo upload.

The "Get Folder" intent is used as the basis for the "Save File" intent, as this means that a new online check of the specified parent folder does not have to be executed multiple times. This is because at least one online enumeration is required to ensure that this folder is in the FileProviderExtension database. Since the subsequent call of `importDocument(at:toParentItemIdentifier)` requires the parentItemIdentifier, i.e. the identifier of the specified folder.

The Save File intent calls `importDocument(at:toParentItemIdentifier)` via XPC for the selected vault and is considered successful once the file has been imported locally and the upload has been scheduled. Thus, this behaves analogously to manually importing a file via the Files app. This is important to ensure a fast runtime of the single intent, as they have a limited execution time.

### CODE REVIEW

File: SharedResources/en.lproj/Localizable.strings [hunk 1]
```diff
@@ -106,6 +106,14 @@
 "fileProvider.uploadProgress.title" = "Uploading…";
 "fileProvider.uploadProgress.missingDomainError" = "Could not find domain.";
 
+"getFolderIntent.error.missingPath" = "No path was provided. Please provide a valid path for which a folder should be returned.";
+"getFolderIntent.error.noVaultSelected" = "No Vault has been selected.";
```
Comment (tobihagemann) [inline SharedResources/en.lproj/Localizable.strings:110]:
```suggestion
"getFolderIntent.error.noVaultSelected" = "No vault has been selected.";
```

File: Cryptomator.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -3123,9 +3261,74 @@
...
+				SWIFT_EMIT_LOC_STRINGS = YES;
+				SWIFT_VERSION = 5.0;
+				TARGETED_DEVICE_FAMILY = "1,2";
+			};
+			name = Release;
+		};
 		4AE97DD024572E4A00452814 /* Debug */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
+				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = YES;
```
Comment (tobihagemann) [inline Cryptomator.xcodeproj/project.pbxproj:3331]: Can the lines 3331 to 3407 be reverted?

**Review by tobihagemann — CHANGES_REQUESTED** [review]
Haven't tested it yet, will review it tomorrow furthermore. Looking forward to it! Just wanted to request some minor changes for now that I've noticed briefly.

**Review by tdung — APPROVED** [review]

File: Cryptomator.xcodeproj/project.pbxproj (hunk 2 above)
Comment (phil1995) [reply Cryptomator.xcodeproj/project.pbxproj]: Resolved by 21376d6e1327a6223c3af9dae6b150b4531d821e

File: SharedResources/en.lproj/Localizable.strings [hunk 3]
```diff
@@ -100,6 +100,7 @@
 "fileProvider.error.unlockButton" = "Unlock";
 "fileProvider.clearFileFromCache.title" = "Clear File from Cache";
 "fileProvider.clearFileFromCache.message" = "This only removes the local file from your device and does not delete the file in the cloud.";
+"fileProvider.fileImporting.error.missingPremium" = "Files can only be imported if write access is available for the vault. To get write access, unlock the full version in the main app.";
```
Comment (tobihagemann) [inline SharedResources/en.lproj/Localizable.strings:103]:
```suggestion
"fileProvider.fileImporting.error.missingPremium" = "Unlock the full version in the Cryptomator app to gain write access to your vaults.";
```

**Review by tobihagemann — CHANGES_REQUESTED** [review]

**Review by tobihagemann — APPROVED** [review]

**Review by tdung — COMMENTED** [review]
Y

**Comment by micha-09** [comment]
Do you have an Shortcut example, how it works? I don't know the correct parameter in path for Get Folder.

**Comment by tobihagemann** [comment]
We've written a guide here: https://docs.cryptomator.org/en/1.6/ios/shortcuts-guide/

The link is also available in the settings of Cryptomator. Hope this helps!

**Comment by micha-09** [comment]
Perfect thank 👍

**Comment by micha-09** [comment]
Are there any known issues while uploading pictures with shortcuts? The date (modified and created) of uploaded pictures is set to "upload date"?

**Comment by tobihagemann** [comment]
The only known issue is also documented in the Shortcuts Guide. Uploading a large amount of photos at once is very slow so it's better to set a limit.

I wouldn't have considered the modification date to be an issue but you can certainly open a new one. I would even say that this is a feature request because the current feature is working as intended.

**Comment by XxBlackHatxX** [comment]
I want to give an overwhelming thanks to your team for doing this. Adding shortcuts support allows a lot of customizations for different workflows. So far this has been working great for me. I even have an automation where it runs the shortcut anytime I plug in my charger. I was also able to create my own automation shortcut that was able to transfer my files over from Google Drive and into Cryptomator automatically.

I have been using the current actions but they require your vault to be unlocked before hand. It would be useful to have Shortcut actions for the following.
1. Shortcut action to check a select vaults status (ie locked or unlocked) and return the current state. 
2. Shortcut action to deep link right into the vault I select (i.e. what ever link you use on the "Open in Files app" button in the app)
3. Shortcut action to lock a selected vault.
4. Shortcut action to unlock a selected vault. If that is not possible no.2 would be a great work around for that as well since when you use the deep link to the files app it asks you to unlock the vault.


I will put in a new feature request for these actions. Thank you again this app has been amazing.

**Pull request merged**